tdp_core/sql.py

```diff
@@ -18,7 +20,7 @@ def load_ids(idtype, mapping):
 
 
 @app.route('/')
```

Comment:
Calling the API route http://localhost:8080/api/tdp/db/ shows an _Internal Server_ error with the following stack trace for independent of the login status:

```
api_1          | Traceback (most recent call last):         
api_1          |   File "/usr/local/lib/python2.7/site-packages/gevent/pywsgi.py", line 884, in handle_one_response     
api_1          |     self.run_application()
api_1          |   File "/usr/local/lib/python2.7/site-packages/geventwebsocket/handler.py", line 88, in run_application
api_1          |     return super(WebSocketHandler, self).run_application()                                             
api_1          |   File "/usr/local/lib/python2.7/site-packages/gevent/pywsgi.py", line 870, in run_application         
api_1          |     self.result = self.application(self.environ, self.start_response)                                  
api_1          |   File "/usr/local/lib/python2.7/site-packages/werkzeug/contrib/fixers.py", line 152, in __call__      
api_1          |     return self.app(environ, start_response)                                                           
api_1          |   File "phovea_server/phovea_server/dispatcher.py", line 63, in __call__                               
api_1          |     return app(environ, start_response)    
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 2000, in __call__                   
api_1          |     return self.wsgi_app(environ, start_response)                                                      
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1991, in wsgi_app                   
api_1          |     response = self.make_response(self.handle_exception(e))                                            
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1567, in handle_exception           
api_1          |     reraise(exc_type, exc_value, tb)       
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app                   
api_1          |     response = self.full_dispatch_request()
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1641, in full_dispatch_request      
api_1          |     rv = self.handle_user_exception(e)     
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1544, in handle_user_exception      
api_1          |     reraise(exc_type, exc_value, tb)       
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1639, in full_dispatch_request      
api_1          |     rv = self.dispatch_request()           
api_1          |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1625, in dispatch_request           
api_1          |     return self.view_functions[rule.endpoint](**req.view_args)                                         
api_1          |   File "/phovea/tdp_core/tdp_core/security.py", line 11, in decorated_view                             
api_1          |     if kwargs['view_name'] is not None and kwargs['database'] is not None:                             
api_1          | KeyError: 'view_name'     
api_1          | {'GATEWAY_INTERFACE': 'CGI/1.1',           
api_1          |  'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3',
api_1          |  'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',                                                          
api_1          |  'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.9', 
api_1          |  'HTTP_CACHE_CONTROL': 'max-age=0',        
api_1          |  'HTTP_CONNECTION': 'close',               
api_1          |  'HTTP_COOKIE': 'randomCredentials=sleepy_joliot@ifbqth; cookiebar=CookieAllowed; remember_token=sleepy_joliot|ec27800dcdfb9c50c9c52ad2429877de86bb217263c77f65c1ede547eab9722339492d4f320a9224aab6372d3afe2d6ddedefd2d03dd21543b418b653f0dbfb2; _pk_id.3.1fff=9f83c292cec1dde4.1563276156.4.1565789866.1565789072.; session=eyJfZnJlc2giOmZhbHNlLCJ1c2VyX2lkIjoic2xlZXB5X2pvbGlvdCJ9.EDv4sA.H-NMXOgZ2TT__-aypMra5nn4LcE',
api_1          |  'HTTP_HOST': 'localhost:8080',            
api_1          |  'HTTP_SEC_FETCH_MODE': 'navigate',        
api_1          |  'HTTP_SEC_FETCH_SITE': 'none',            
api_1          |  'HTTP_SEC_FETCH_USER': '?1',              
api_1          |  'HTTP_UPGRADE_INSECURE_REQUESTS': '1',    
api_1          |  'HTTP_USER_AGENT': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36',       
api_1          |  'PATH_INFO': '/',        
api_1          |  'QUERY_STRING': '',      
api_1          |  'REMOTE_ADDR': '172.26.0.1',              
api_1          |  'REMOTE_PORT': '38452',  
api_1          |  'REQUEST_METHOD': 'GET', 
api_1          |  'SCRIPT_NAME': '/api/tdp/db',             
api_1          |  'SERVER_NAME': 'dc6701a26e09',            
api_1          |  'SERVER_PORT': '80',     
api_1          |  'SERVER_PROTOCOL': 'HTTP/1.1',            
api_1          |  'SERVER_SOFTWARE': 'gevent/1.1 Python/2.7',                                                           
api_1          |  'werkzeug.proxy_fix.orig_http_host': 'localhost:8080',                                                
api_1          |  'werkzeug.proxy_fix.orig_remote_addr': '172.26.0.1',                                                  
api_1          |  'werkzeug.proxy_fix.orig_wsgi_url_scheme': 'http',                                                    
api_1          |  'werkzeug.request': <Request 'http://localhost:8080/api/tdp/db/' [GET]>,                              
api_1          |  'wsgi.errors': <open file '<stderr>', mode 'w' at 0x7f9bc2ce71e0>,                                    
api_1          |  'wsgi.input': <gevent.pywsgi.Input object at 0x7f9ba84af050>,                                         
api_1          |  'wsgi.multiprocess': False,               
api_1          |  'wsgi.multithread': False,                
api_1          |  'wsgi.run_once': False,  
api_1          |  'wsgi.url_scheme': 'http',                
api_1          |  'wsgi.version': (1, 0)} failed with KeyError                                                          
api_1          |                                                    
```

Reply:
please pull again. this should be fixed with my last commit

Reply:
at least it works for me

Reply:
see also https://github.com/datavisyn/tdp_core/pull/191/commits/73e87e810cc43156d1584bd45cd004c139d1c3fc

Reply:
it works for me in my app --> @thinkh can you please check it again

Reply:
Yes, now it works for me.